methods.py

```diff
@@ -903,160 +904,109 @@ def using_emcc(env):
+    if env.GetOption("clean") or not env["SCONS_CACHE"]:
+        return
 
-        def delete(self, files):
-            if len(files) == 0:
-                return
-            if env["verbose"]:
-                # Utter something
-                print("Purging %d %s from cache..." % (len(files), "files" if len(files) > 1 else "file"))
-            [os.remove(f) for f in files]
-
-        def file_list(self):
-            if self.path is None:
-                # Nothing to do
-                return []
-            # Gather a list of (filename, (size, atime)) within the
-            # cache directory
-            file_stat = [(x, os.stat(x)[6:8]) for x in glob.glob(os.path.join(self.path, "*", "*"))]
-            if file_stat == []:
-                # Nothing to do
-                return []
-            # Weight the cache files by size (assumed to be roughly
-            # proportional to the recompilation time) times an exponential
-            # decay since the ctime, and return a list with the entries
-            # (filename, size, weight).
-            current_time = time.time()
-            file_stat = [(x[0], x[1][0], (current_time - x[1][1])) for x in file_stat]
-            # Sort by the most recently accessed files (most sensible to keep) first
-            file_stat.sort(key=lambda x: x[2])
-            # Search for the first entry where the storage limit is
-            # reached
-            sum, mark = 0, None
-            for i, x in enumerate(file_stat):
-                sum += x[1]
-                if sum > self.limit:
-                    mark = i
-                    break
-            if mark is None:
-                return []
-            else:
-                return [x[0] for x in file_stat[mark:]]
+    files = glob.glob(os.path.join(env["SCONS_CACHE"], "*", "*"))
+    if not files:
+        return
 
-    def cache_finally():
-        nonlocal cleaner
+    # Remove all text files, store binary files in list of (filename, size, atime).
+    purge = []
+    stats = []
+    for file in files:
+        # A bit of an antipattern, but failing a utf-8 decode is the easiest way to determine if
+        # a file is binary.
         try:
-            cleaner.clean()
-        except Exception:
-            pass
-
-    cache_directory = os.environ.get("SCONS_CACHE")
-    # Simple cache pruning, attached to SCons' progress callback. Trim the
-    # cache directory to a size not larger than cache_limit.
-    cache_limit = float(os.getenv("SCONS_CACHE_LIMIT", 1024)) * 1024 * 1024
-    cleaner = cache_clean(cache_directory, cache_limit)
+            with open(file, encoding="utf-8") as out:
+                out.read(100)
+            purge.append(file)
+        except UnicodeDecodeError:
+            stats.append((file, *os.stat(file)[6:8]))
+
+    # Sort by most recent access (most sensible to keep) first. Search for the first entry where
+    # the cache limit is reached.
+    stats.sort(key=lambda x: x[2], reverse=True)
```

Comment:
You've changed the sorting behavior here, it now sorts bh the file time ascending, which is backwards, unless I've missed something here

Reply:
It seems that way at a glance, but it originally performed `current_time - atime` before sorting. This was a remnant of when the logic was more complex, so the operation is functionally `-atime` because the first value is a constant. Because reversing the entries is possible from `sort()` directly, the operation could be removed entirely while achieving the same result

Reply:
Ah sorry missed the `reverse=True` not familiar with the sorting in python! My bad!